examples/ota-provider-app/ota-provider-common/DefaultUserConsentProvider.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <ota-provider-common/DefaultUserConsentProvider.h>
+
+namespace chip {
+namespace ota {
+
+DefaultUserConsentProvider::DefaultUserConsentProvider()
+{
+    for (uint8_t i = 0; i < kMaxUserConsentEntries; i++)
+    {
+        mUserConsentEntries[i].nodeId = kUndefinedNodeId;
+    }
+}
+
+UserConsentState DefaultUserConsentProvider::GetUserConsentState(NodeId nodeId, EndpointId endpoint, uint32_t currentVersion,
+                                                                 uint32_t newVersion)
+{
+    for (uint8_t i = 0; i < kMaxUserConsentEntries; i++)
+    {
+        if (mUserConsentEntries[i].nodeId == nodeId && mUserConsentEntries[i].endpoint == endpoint)
+        {
+            return mUserConsentEntries[i].state;
+        }
+    }
+    return UserConsentState::kGranted;
+}
+
+CHIP_ERROR DefaultUserConsentProvider::SetUserConsentState(NodeId nodeId, EndpointId endpoint, UserConsentState state)
```

Comment:
See comment above re: more arguments.

Overall, it would be important to be able to get an iterator over all current pending requests.

BTW, this entire implementation, as constructed, just "grows unboundedly". It's fine for an example, but hopefully is not required to be in this form outside of examples.

Reply:
Using `UserConsentSubject` here.